src/sha256.c

```diff
@@ -0,0 +1,180 @@
+// Copyright 2023 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string.h>
+
+#include "rcutils/sha256.h"
+
+static inline uint32_t rotright(uint32_t a, uint8_t b)
+{
+  return (a >> b) | (a << (32 - b));
```

Comment:
The `32 - b` is slightly suspect.  What if `b` is 255?  We would then end up shifting left with a negative number, which I think is undefined behavior.  I guess given how we use this function it won't happen, but maybe we should add an assert or something?  What do you think?

Reply:
Yes, this is really a `uint5_t` expected input range. I think an assert is fine as long as we're optimizing them out in release builds.